src/WebJobs.Script.WebHost/App_Start/AutofacBootstrap.cs

```diff
@@ -13,16 +14,15 @@ internal static void Initialize(ScriptSettingsManager settingsManager, Container
         {
             builder.RegisterInstance(settingsManager);
 
-            // register the resolver so that it is disposed when the container
-            // is disposed
-            var webHostResolver = new WebHostResolver(settingsManager);
-            builder.RegisterInstance(webHostResolver);
+            builder.RegisterType<WebHostResolver>().SingleInstance();
```

Comment:
Some of these changes are outside of the scope of the feature this PR introduces. I'm leaving this here for feedback, but we can revert a lot of these without problems.